### ISSUE TITLE
SA1000: Generic constraint new()

### ISSUE THREAD

**Issue by pdelvo** [DotNetAnalyzers/StyleCopAnalyzers#340]
Currently SA1000 is reporting a diagnostic for this:
```csharp
public void Foo<T>() where T : IInterface, new()
{
    // ...
}
```
SA1000 is reporting, that new has to be followed by a space. If you add this space (new ()), then you get a SA1008. So there is currently no way to get a generic ctor constraint without violating a rule.
